crates/vm/levm/src/vm.rs

```diff
@@ -1303,6 +1371,226 @@ impl VM {
+
+            let Ok(authority) = libsecp256k1::recover(&message, &signature, &recovery_id) else {
+                continue;
+            };
+
+            let public_key = authority.serialize();
+            let mut hasher = Keccak256::new();
+            hasher.update(
+                public_key
+                    .get(1..)
+                    .ok_or(VMError::Internal(InternalError::SlicingError))?,
+            );
+            let address_hash = hasher.finalize();
+
+            // Get the last 20 bytes of the hash -> Address
+            let authority_address_bytes: [u8; 20] = address_hash
+                .get(12..32)
+                .ok_or(VMError::Internal(InternalError::SlicingError))?
+                .try_into()
+                .map_err(|_| VMError::Internal(InternalError::ConversionError))?;
+            let authority_address = Address::from_slice(&authority_address_bytes);
+
+            // 4. Add authority to accessed_addresses (as defined in EIP-2929).
+            self.accrued_substate
+                .touched_accounts
+                .insert(authority_address);
+            let authority_account_info = self.get_account_no_push_cache(authority_address).info;
+
+            // 5. Verify the code of authority is either empty or already delegated.
+            let empty_or_delegated = authority_account_info.bytecode.is_empty()
+                || has_delegation(&authority_account_info)?;
+            if !empty_or_delegated {
+                continue;
+            }
+
+            // 6. Verify the nonce of authority is equal to nonce. In case authority does not exist in the trie, verify that nonce is equal to 0.
+            // If it doesn't exist, it means the nonce is zero. The access_account() function will return AccountInfo::default()
+            // If it has nonce, the account.info.nonce should equal auth_tuple.nonce
+            if authority_account_info.nonce != auth_tuple.nonce {
+                continue;
+            }
+
+            // 7. Add PER_EMPTY_ACCOUNT_COST - PER_AUTH_BASE_COST gas to the global refund counter if authority exists in the trie.
+            // CHECK: we don't know if checking the cache is correct. More gas tests pass but the set_code_txs tests went to half.
+            if self.db.account_exists(authority_address)
+                || self.cache.contains_key(&authority_address)
+            {
+                let refunded_gas_if_exists = PER_EMPTY_ACCOUNT_COST - PER_AUTH_BASE_COST;
+                refunded_gas = refunded_gas
+                    .checked_add(refunded_gas_if_exists)
+                    .ok_or(VMError::Internal(InternalError::GasOverflow))?;
+            }
+
+            // 8. Set the code of authority to be 0xef0100 || address. This is a delegation designation.
+            let mut delegation_bytes = Vec::new();
+            delegation_bytes.extend_from_slice(&SET_CODE_DELEGATION_BYTES);
+            delegation_bytes.extend_from_slice(auth_tuple.address.as_bytes());
```

Comment:
Nit. You can use `concat` here.
```suggestion
            let delegation_bytes = [&SET_CODE_DELEGATION_BYTES, auth_tuple.address.as_bytes()].concat();
```